packages/notebook-extension/src/index.ts

```diff
@@ -3368,6 +3378,156 @@ function addCommands(
+      }
+    },
+
+    isEnabled: args => {
+      const current = getCurrent(tracker, shell, { ...args, activate: false });
+      if (!current) {
+        return false;
+      }
+
+      const editor = current.content.activeCell?.editor;
+      if (editor) {
+        const selection = editor.getSelection();
+        const hasEditorSelection =
+          selection.start.line !== selection.end.line ||
+          selection.start.column !== selection.end.column;
+        if (hasEditorSelection) {
+          return true;
+        }
+      }
+
+      // Check for text selection in output area
+      const domSelection = window.getSelection();
+      return !!domSelection && domSelection.toString().trim().length > 0;
+    },
+    describedBy: {
+      args: {
+        type: 'object',
+        properties: {}
+      }
+    }
+  });
+
+  commands.addCommand(CommandIDs.cutSelectedtext, {
+    label: trans.__('Cut Selected Text'),
+    caption: trans.__('Cut selected text from the active cell or output area'),
+    execute: async args => {
+      const current = getCurrent(tracker, shell, args);
+      if (!current) {
+        return;
+      }
+
+      const editor = current.content.activeCell?.editor;
+      if (editor) {
+        const selection = editor.getSelection();
+        const start = editor.getOffsetAt(selection.start);
+        const end = editor.getOffsetAt(selection.end);
+        const text = editor.model.sharedModel.getSource().slice(start, end);
+        if (text) {
+          await navigator.clipboard.writeText(text);
+          editor.replaceSelection?.('');
+          return;
+        }
+      }
+    },
+
+    isEnabled: args => {
+      const current = getCurrent(tracker, shell, { ...args, activate: false });
+      if (!current) {
+        return false;
+      }
+
+      const editor = current.content.activeCell?.editor;
+      if (!editor) {
+        return false;
+      }
+
+      const selection = editor.getSelection();
+      const hasEditorSelection =
+        selection.start.line !== selection.end.line ||
+        selection.start.column !== selection.end.column;
+      return hasEditorSelection;
+    },
+    describedBy: {
+      args: {
+        type: 'object',
+        properties: {}
+      }
+    }
+  });
+
+  commands.addCommand(CommandIDs.pasteText, {
+    label: trans.__('Paste Text'),
+    caption: trans.__(
+      'Paste text from the clipboard into the active cell editor'
+    ),
+    execute: async args => {
+      const current = getCurrent(tracker, shell, args);
+      if (!current) {
+        return;
+      }
+
+      const editor = current.content.activeCell?.editor;
+      if (!editor) {
+        return;
+      }
+
+      try {
+        const text = await navigator.clipboard.readText();
+        if (text) {
+          editor.replaceSelection?.(text);
+        }
+      } catch (err) {
+        // browser limitation fallback (e.g Firefox)
+        Clipboard.showPasteUnavailableDialog(trans);
+      }
```

Comment:
Maybe all of this could be in a helper function in Clipboard? Like `Clipboard.paste(editor)`?